src/mux-analytics.brs

```diff
@@ -535,6 +584,122 @@ function muxAnalytics() as Object
     end if
   end sub
 
+  prototype.videoContentIndexChangeHandler = sub(contentIndex as integer)
+    if contentIndex > 0
+      m._addEventToQueue(m._createEvent("ended"))
+      m._endView(true)
+      m._startView(true)
+      m._triggerPlayEvent()
+      if m._Flag_atLeastOnePlayEventForContent = false
+        if m._viewTimeToFirstFrame = Invalid
+          if m._viewStartTimestamp <> Invalid and m._viewStartTimestamp <> 0
+            date = m._getDateTime()
+            now = 0# + date.AsSeconds() * 1000.0# + date.GetMilliseconds()
+            m._viewTimeToFirstFrame = now - m._viewStartTimestamp
+          end if
+        end if
+      end if
+      m._addEventToQueue(m._createEvent("playing"))
+    end if
+  end sub
+
+  prototype.videoStreamingSegmentChangeHandler = sub(videoSegment as object)
+    if videoSegment <> Invalid
+      if m._lastPlayerWidth <> Invalid AND m._lastPlayerHeight <> Invalid AND m._lastPlayheadPosition <> Invalid AND m._lastSourceWidth <> Invalid AND m._lastSourceHeight <> Invalid AND videoSegment.segStartTime <> Invalid
+        player_playhead_time = Int(videoSegment.segStartTime * 1000)
+        if m._lastPlayerWidth >= 0 AND m._lastPlayerHeight >= 0 AND m._lastPlayheadPosition >= 0 AND player_playhead_time >= 0 AND m._lastSourceWidth > 0 AND m._lastSourceHeight > 0
+          timeDiff = player_playhead_time - m._lastPlayheadPosition
+          scale = m._min(m._lastPlayerWidth / m._lastSourceWidth, m._lastPlayerHeight / m._lastSourceHeight)
+          upscale = m._max(0, scale - 1)
+          downscale = m._max(0, 1 - scale)
+          m._viewMaxUpscalePercentage = m._max(m._viewMaxUpscalePercentage, upscale)
```

Comment:
given that our `_max` etc functions treat invalid as 0, is there a chance that we set this to 0 when we really should never set it? for instance, we shouldn't set this unless we _know_ it wasn't upscaled or downscaled

that might not be a legitimate concern given all of the other checks we have, though

Reply:
Good point. In that case, I can modify the max/min functions. So if any value is invalid, it will return the other value by default. any thought?

Reply:
If any of those values (`_lastPlayerWidth`, `_lastPlayerHeight`, `_lastSourceWidth`, or `_lastSourceHeight`) are invalid, or if the division of those are invalid, we'd want upscale/downscale to also be invalid so they're not included in the beacons. I think this case is all covered above by making sure they're set, and sourceWidth/Height are greater than zero, but just wanted to make sure

Reply:
Yes, I've covered all the conditions you mentioned but I haven't set upscale/downscale to invalid. So in that case, beacons will still include the last values. I can reset those values to invalid if required.